java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Instance.java

```diff
@@ -82,15 +82,19 @@ protected void afterExecute(Runnable t, Throwable e) {
         private final ThreadObserverHelper _observerHelper;
     }
 
+    // The thead pool executor uses an unbounded queue. The tasks would wait in the queue until a
+    // core pool thread is available to run it and the value of the maximumPoolSize therefore
+    // doesn't have any effect.
```

Comment:
The comment is not clear.

Presumably, maximumPoolSize has not effect because the executor uses an unbounded queue? As written, it sounds like a consequence of the wait?

It's also not clear to me what this means / add:
> The tasks would wait in the queue until a core pool thread is available to run it 

This is not obvious?